### PR TITLE
feat: make aspnetcore chart usable for non-.NET workloads (FENG-2321)

### PR DESCRIPTION
**Pull request by samrobillard**
Jira issue link: [FENG-2321](https://workleap.atlassian.net/browse/FENG-2321)

## Summary

Adds two additive, backward-compatible knobs to the `aspnetcore` chart so non-.NET workloads (e.g. the TypeScript BFF `wl-app-server`) can use it without forking:

- **`image.containerPort`** (default `8080`) — parameterizes the container port and the rendered `ASPNETCORE_URLS` value, so a non-default port works for both ASP.NET and non-ASP.NET workloads.
- **`aspnetcore.injectEnvVars`** (default `true`) — gates injection of `DOTNET_ENVIRONMENT` and `ASPNETCORE_URLS`. Set to `false` for non-.NET workloads.

Defaults preserve existing behavior — current ASP.NET consumers see no change.

The `env:` block is now omitted entirely when `injectEnvVars=false` and `extraEnvVars` is empty, so we don't render an invalid empty list.

Chart bumped to **5.1.0** (MINOR, additive).

Jira: [FENG-2321](https://workleap.atlassian.net/browse/FENG-2321)

## Why granular over a single `aspnet: true/false` flag

Couples two unrelated decisions (port number vs. env injection). An ASP.NET app on a non-default Kestrel port should not have to disable env injection just to override the port. Granular knobs also avoid a future rename when the chart name no longer matches its usage.

## Coverage

- Five new helm-unittest cases in `tests/deployment_test.yaml` covering: default port, custom port (with `ASPNETCORE_URLS` propagation), default env var injection, env var suppression, and the empty-env edge case.
- Four new Makefile schema tests for `image.containerPort` (min/max/valid) and `aspnetcore.injectEnvVars=false` rendering.
- New CI step in `ci-aspnetcore.yml` that deploys `tests/values-nondotnet.yaml` (Node/TypeScript `mendhak/http-https-echo:40` on port 3000, env injection off) to each Kind cluster in the matrix and asserts no `DOTNET_*` env vars leak into the running pod.

## Test plan

- [x] `helm lint --strict ./charts/aspnetcore/` passes
- [x] `helm unittest charts/aspnetcore` — 105 passed
- [x] `CI=true make tests` — all schema and precheck tests pass (incl. new ones)
- [x] `helm template` diff shows only the `helm.sh/chart` label changing (5.0.0 → 5.1.0); deployment env block / port unchanged for default values
- [x] `helm template -f tests/values-nondotnet.yaml` renders `containerPort: 3000` and only `HTTP_PORT` env var
- [x] Schema rejects `image.containerPort=0` and `=70000`
- [ ] CI Kind matrix (v1.27–v1.30) deploys both default and non-.NET scenarios

## Out of scope

- Non-Deployment workload kinds (Job/StatefulSet/etc.)
- Multi-container / sidecar support
- Chart rename / split

[FENG-2321]: https://workleap.atlassian.net/browse/FENG-2321?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -201,6 +201,28 @@ tests/schema/autoscaling-cpu-target-too-high: export EXPECTED_ERROR_MESSAGE="(ta
...
+
+# Test image.containerPort accepts a non-default valid value
+tests/schema/image-containerport-valid: export TEST_DISPLAY_NAME="Schema should accept a custom image.containerPort"
+tests/schema/image-containerport-valid:
+	@${HELM_TEMPLATE} --set image.containerPort=3000 ${SHOULD_SUCCEED_AND_THEN} ${DISPLAY_RESULT}
+
+# Test aspnetcore.injectEnvVars=false suppresses DOTNET_* env vars
+tests/schema/aspnetcore-injectenvvars-false: export TEST_DISPLAY_NAME="Disabling injectEnvVars should suppress DOTNET_* env vars"
+tests/schema/aspnetcore-injectenvvars-false:
+	@${HELM_TEMPLATE} --set aspnetcore.injectEnvVars=false 2>&1 | grep -q -v 'DOTNET_ENVIRONMENT' && ${DISPLAY_RESULT}
```
[the remark this answers is not in the document]
Comment (samrobillard) [reply Makefile]: Fixed in 8a15b47. Capturing the rendered output first (`OUT="$(helm template ... 2>&1)"`) so a templating failure propagates via the `&&` chain, then asserting non-presence with `! echo "$OUT" | grep -q "DOTNET_ENVIRONMENT"`. Verified locally with three scenarios: helm template failure → fails; DOTNET_ENVIRONMENT injected via extraEnvVars → fails; correct case → passes.

**Review by mcharbo — APPROVED** [review]
